### PR TITLE
Closes #2443 - Read/Write SegArray of Strings HDF5

### LINKED ISSUE #2443: Read/Write SegArray of Strings to HDF5
Once SegArrays containing Strings are fully supported, HDF5 write and reads will need to be updated to handle SegArray of Strings.

### PR DESCRIPTION
**Pull request by Ethan-DeBandi99**
Closes #2443 

This PR adds support for reading and writing SegArrays with Strings value objects.

### CODE REVIEW

File: src/HDF5Msg.chpl [hunk 1]
```diff
@@ -1207,9 +1228,83 @@ module HDF5Msg {
...
+    }
 
-                writeArkoudaMetaData(file_id, group, objType, dtype);
+    /*
+    * Write SegArray containing Strings values to HDF5
+    * String offsets are localized to SegArray offsets. The String values are then localized to ensure entire
+    * value of the string can be read.
+    */
+    proc writeNestedSegmentedDistDset(filenames: [] string, group: string, objType: string, overwrite: bool, values, segments, st: borrowed SymTab, type t) throws {
+        // TODO - better variable naming
```
Comment (pierce314159) [inline src/HDF5Msg.chpl:1243]: do we still need this todo?

File: tests/io_test.py [hunk 2]
```diff
@@ -1199,6 +1205,19 @@ def test_overwrite_single_dset(self):
 
             self.assertEqual(f1_size, f2_size)
 
+    def test_segarray_str_hdf5(self):
+        language = ak.array(["english", "spanish"])
```
Comment (pierce314159) [inline tests/io_test.py:1209]: it seems we don't really use `language` anywhere

**Review by pierce314159 — APPROVED** [review]
Looks good to me!

**Review by jaketrookman — APPROVED** [review]
Looks good

File: src/HDF5Msg.chpl (hunk 1 above)
Comment (Ethan-DeBandi99) [reply src/HDF5Msg.chpl:1243]: No this was left over. It's been removed.

File: tests/io_test.py (hunk 2 above)
Comment (Ethan-DeBandi99) [reply tests/io_test.py:1209]: Yeah that was from another test I did. Removed.

**Review by pierce314159 — APPROVED** [review]

**Review by jaketrookman — APPROVED** [review]
Looks good

**Pull request merged**